pkg/cgroup/cgroup_stats_linux.go

```diff
@@ -0,0 +1,145 @@
+//go:build linux
+// +build linux
+
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cgroup
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containerd/cgroups"
+	statsv1 "github.com/containerd/cgroups/stats/v1"
+	"github.com/containerd/cgroups/v3/cgroup2"
+	statsv2 "github.com/containerd/cgroups/v3/cgroup2/stats"
+
+	"github.com/sustainable-computing-io/kepler/pkg/config"
+)
+
+type CCgroupStatHandler struct {
+	cgroupV1StatHandler *statsv1.Metrics
+	cgroupV2StatHandler *statsv2.Metrics
+}
+
+var (
+	AvailableCGroupMetrics = []string{
+		config.CgroupfsMemory, config.CgroupfsKernelMemory, config.CgroupfsTCPMemory,
+		config.CgroupfsCPU, config.CgroupfsSystemCPU, config.CgroupfsUserCPU,
+		config.CgroupfsReadIO, config.CgroupfsWriteIO, config.BlockDevicesIO,
+	}
+)
+
+func NewCGroupStatHandler(pid int) (*CCgroupStatHandler, error) {
+	hander := CCgroupStatHandler{}
+	p := fmt.Sprintf(procPath, pid)
+	_, path, err := cgroups.ParseCgroupFileUnified(p)
+	if err != nil {
+		return nil, err
+	}
+
+	if config.GetCGroupVersion() == 1 {
+		manager, err := cgroups.Load(cgroups.V1, cgroups.StaticPath(path))
+		if err != nil {
+			return &hander, err
+		}
+		v1StatHandler, err := manager.Stat(cgroups.IgnoreNotExist)
+		if err != nil {
+			return &hander, err
+		}
+		hander.cgroupV1StatHandler = v1StatHandler
+	} else {
```

Comment:
not sure worthy a quick comment on what we are doing here ..

Reply:
@jichenjc I included a comment in the function.